### PR TITLE
Wsp/sequence

### PR DESCRIPTION
**Pull request by wpride**
Add sequenceId checking and incrementing to Formplayer

### CODE REVIEW

File: src/main/java/application/FormController.java [hunk 1]
```diff
@@ -76,36 +76,36 @@ public NewFormSessionResponse openIncompleteForm(@RequestBody IncompleteSessionR
...
                 answerQuestionBean.getAnswer() != null? answerQuestionBean.getAnswer().toString() : null,
                 answerQuestionBean.getFormIndex());
-
+        formEntrySession.setSequenceId(formEntrySession.getSequenceId() + 1);
         updateSession(formEntrySession, session);
 
-        AnswerQuestionResponseBean responseBean = mapper.readValue(resp.toString(), AnswerQuestionResponseBean.class);
-        responseBean.setSequenceId(formEntrySession.getSequenceId() + 1);
+        FormEntryResponseBean responseBean = mapper.readValue(resp.toString(), FormEntryResponseBean.class);
+        responseBean.setSequenceId(formEntrySession.getSequenceId());
```
Comment (benrudolph) [inline src/main/java/application/FormController.java]: why is the `+ 1` removed here?

File: src/main/java/application/FormController.java [hunk 2]
```diff
@@ -162,38 +162,32 @@ public EvaluateXPathResponseBean evaluateXpath(@RequestBody EvaluateXPathRequest
...
-
         JSONObject resp = JsonActionUtils.deleteRepeatToJson(formEntrySession.getFormEntryController(),
                 formEntrySession.getFormEntryModel(),
-                repeatRequestBean.getRepeatIndex(), repeatRequestBean.getFormIndex());
-
+                deleteRepeatRequestBean.getRepeatIndex(), deleteRepeatRequestBean.getFormIndex());
         updateSession(formEntrySession, serializableFormSession);
-
-        return mapper.readValue(resp.toString(), RepeatResponseBean.class);
+        formEntrySession.setSequenceId(formEntrySession.getSequenceId() + 1);
```
Comment (benrudolph) [inline src/main/java/application/FormController.java]: it'd almost be nice to have this operate at a slightly higher level as some sort of decorator or filter

File: src/main/java/application/FormController.java [hunk 3]
```diff
@@ -236,7 +230,8 @@ public GetSessionsResponse getSessions(@RequestBody GetSessionsBean getSessionRe
     private void updateSession(FormSession formEntrySession, SerializableFormSession serialSession) throws IOException {
         serialSession.setFormXml(formEntrySession.getFormXml());
         serialSession.setInstanceXml(formEntrySession.getInstanceXml());
-        serialSession.setSequenceId(formEntrySession.getSequenceId() + 1);
+        serialSession.setSequenceId(formEntrySession.getSequenceId());
+        System.out.println("Increment sequence id from: " + formEntrySession.getSequenceId());
```
Comment (benrudolph) [inline src/main/java/application/FormController.java]: print

File: src/main/java/application/FormController.java (hunk 1 above)
Comment (wpride) [inline src/main/java/application/FormController.java]: setting it in the session object above (line 87) instead

**Comment by benrudolph** [comment]
👍 on ✔️

**Pull request merged**